neuralprophet/df_utils.py

```diff
@@ -1331,27 +1313,75 @@ def infer_frequency(df, freq, n_lags, min_freq_percentage=0.7):
     return freq_str
 
 
-def compare_dict_keys(dict_1, dict_2, name_dict_1, name_dict_2):
-    """Compare keys of two different dicts (i.e., events and dataframes).
+def compare_df_ids(df_1, dict_2, name_df_1, name_df_2):
+    """Compare ids of original df and features (i.e., events, regressors, etc).
 
     Parameters
     ----------
-        dict_1 : dict
-            first dict
+        df_1 : pd.DataFrame
+            df
         dict_2 : dict
-            second dict
-        name_dict_1 : str
+            dict
+        name_df_1 : str
             name of first dict
-        name_dict_2 : str
+        name_df_2 : str
             name of second dict
 
+
     """
-    df_names_1, df_names_2 = list(dict_1.keys()), list(dict_2.keys())
+    df_names_1, df_names_2 = list(df_1["ID"].unique()), list(dict_2.keys())
     if len(df_names_1) != len(df_names_2):
-        raise ValueError(
-            "Please, make sure {} and {} dicts have the same number of terms".format(name_dict_1, name_dict_2)
-        )
+        raise ValueError("Please, make sure {} and {} dicts have the same number of terms".format(name_df_1, name_df_2))
```

Comment:
@ourownstory I just kept it the way it was before this PR. I can change it so it can receive a different number of events or regressors. Before, we could either provide None, a single df with events or regressors, which would then be applied to all the time series, or a dict with the similar keys from the main time series.